hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/JetSqlResultImplTest.java

```diff
@@ -35,8 +35,8 @@ public void when_closed_then_iteratorFails() {
         JetInstance inst = createJetMember();
         TestBatchSqlConnector.create(inst.getSql(), "m", 1);
         SqlResult sqlResult = inst.getSql().execute("select * from m");
-        sqlResult.close();
         Iterator<SqlRow> iterator = sqlResult.iterator();
+        sqlResult.close();
```

Comment:
I believe this changes was supposed to fix https://github.com/hazelcast/hazelcast-jet/issues/2874?
If that's the case, it does not matter when `close()` is called. The problem is that `JetQueryResultProducer.isDone()` is not called as long as `nextRow != null`.

Reply:
I don't think this was the fix. I think this was because of the interim state in IMDG when the `execute` method was async. If we closed before calling `iterator`, then the `iterator` call failed. But that was reverted so I guess we can revert also this change, will do it.

Issue #2874 was fixed elsewhere, I honestly don't remember where.